src/openrct2/paint/Paint.h

```diff
@@ -270,6 +270,43 @@ struct FootpathPaintInfo
     colour_t SupportColour = 255;
 };
 
+using RotatedImageIndex = std::array<ImageIndex, NumOrthogonalDirections>;
+
+class RotatedTrackPiece
```

Comment:
Have you looked into how many situations this would help out? Every time I've looked into things it always ends up with so many special cases I get put off. 

I don't really see the point of this class it provides nothing over the std::array you could just `using RotatedTrackPiece = RotatedImageIndex`. In general I'm not a fan of inheritance either.